### ISSUE TITLE
Adding WHERE clause to a relation

### ISSUE THREAD

**Issue by salain** [tattersoftware/codeigniter4-relations#28]
HI,

Is there anyway to add a WHERE clause to a relation?
For example retrieve all posts created in 2020 for an Author.

**Comment by MGatner** [comment]
Yes. You can use all the usual Query Builder methods intermixed with Relations methods.

**Comment by salain** [comment]
Sorry, but could you give an example as I can not get it to work.
If I have this and want to filter posts on year 2020:

`$author = $authorModel->with('posts')->where('year', '2020')->findAll();`

I get error column year  not found in where clause

**Comment by MGatner** [comment]
Maybe try `'posts.year'` instead? I cannot dig into this right now but you could try the CodeIgniter forums for help.

**Comment by salain** [comment]
Adding the table name does not work as it adds the where clause to the main query. 
I will try to find a workaround for this.

Thanks

**Comment by MGatner** [comment]
Ahh I see what you mean now. No, this is not currently possible, though we've had requests for handling content from join tables and this would overlap with conditional relationships. I will consider it a feature request.

**Comment by salain** [comment]
Thanks.
I have a basic workaround to suite my tests for now.
